cobalt/h5vcc/h5vcc_settings.cc

```diff
@@ -67,35 +65,69 @@ bool H5vccSettings::Set(const std::string& name, int32 value) const {
   const char kUpdaterMinFreeSpaceBytes[] = "Updater.MinFreeSpaceBytes";
 #endif
 
-  if (name == kDisableMediaCodec &&
-      value < static_cast<int32>(kDisableCodecCombinations.size())) {
-    can_play_type_handler_->SetDisabledMediaCodecs(
-        kDisableCodecCombinations[value]);
+  if (name == kMediaCodecAllowList && value.IsType<std::string>() &&
+      value.AsType<std::string>().size() < 256) {
+    std::vector<std::string> allowed_media_codecs =
+        base::SplitString(value.AsType<std::string>(), ";",
+                          base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+    // Allow only the codecs in the list.
+    // If it is empty string, it allows all codecs.
+    // Examples: input string separated by semicolon
+    // "vp09;vp9": allow only vp9 codec
+    // "": allow all codecs
+    if (allowed_media_codecs.size() > 0) {
+      std::unordered_set<std::string> allowed_media_codecs_set(
+          allowed_media_codecs.begin(), allowed_media_codecs.end());
+      std::vector<std::string> blocked_media_codecs;
+      for (auto& codec : kCodecList) {
+        if (allowed_media_codecs_set.count(codec) <= 0) {
+          blocked_media_codecs.push_back(codec);
+        }
+      }
+      if (blocked_media_codecs.size() < kCodecListSize) {
+        std::string blocked_media_codec_combinations =
+            base::JoinString(blocked_media_codecs, ";");
+        LOG(INFO) << "Codec (" << blocked_media_codec_combinations
+                  << ") is blocked via console/command line.";
+        can_play_type_handler_->SetDisabledMediaCodecs(
```

Comment:
All the code here is to allow us using `SetDisabledMediaCodecs()` to represent an allow list.  We should refactor `SetDisabledMediaCodecs()` to turn it into `SetAllowedMediaCodecs()`.  It's also used to set a command line for blocked codecs, we should also refine the command line to let it accept an allow list of codecs.

Reply:
I agree. Should we split this to another CL for this change?

Reply:
I would say we should use one of the approaches:
1. Use "MediaCodecBlockList" in this CL, refactor it to `MediaCodecAllowList` in a follow up CL, with `SetDisabledMediaCodecs()` replaced by `SetAllowedMediaCodecs()`.
2. Keep using `MediaCodecAllowList` and replace `SetDisabledMediaCodecs()` by `SetAllowedMediaCodecs()` in the this CL.

My reasoning is, we are putting a complex piece of code that will be removed in a follow up CL, where the reviewers and the author have to work together to ensure its correctness.  Removing the code now saves the effort.

Reply:
Keep it as `MediaCodecBlockList`, and will refactor it to `MediaCodecAllowList` in a follow up CL, with `SetDisabledMediaCodecs()` replaced by `SetAllowedMediaCodecs()`.

Reply:
Changed commit message.